android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebViewFragment.java

```diff
@@ -0,0 +1,221 @@
+  private boolean isInstallButton = false;
+  private OnInstallClickedListener callback;
+
+  public void setOnInstallClickedListener(OnInstallClickedListener callback) {
+    this.callback = callback;
+  }
+
+  // This interface to be implemented by calling Activity
+  public interface OnInstallClickedListener {
+    public void onInstallClicked(String pkgTarget, String packageID);
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                           Bundle savedInstanceState) {
+    View v = inflater.inflate(R.layout.fragment_webview_layout, container, false);
+    Bundle bundle = getArguments();
+
+    if (bundle != null) {
+      if (bundle.containsKey("tempPackagePath")) {
+        tempPackagePath = (File)bundle.getSerializable("tempPackagePath");
+      }
+      if (bundle.containsKey("packageID")) {
+        packageID = bundle.getString("packageID");
+      }
+      if (bundle.containsKey("pkgTarget")) {
+        pkgTarget = bundle.getString("pkgTarget");
+      }
+      if (bundle.containsKey("pkgName")) {
+        pkgName = bundle.getString("pkgName");
+      }
+      if (bundle.containsKey("fileName")) {
+        fileName = bundle.getString("fileName");
+      }
+      if (bundle.containsKey("isInstallButton")) {
+        isInstallButton = bundle.getBoolean("isInstallButton");
+      }
+    }
+
+    Toolbar toolbar = (Toolbar) v.findViewById(R.id.titlebar);
+    ((AppCompatActivity)getActivity()).setSupportActionBar(toolbar);
+    ActionBar actionBar = ((AppCompatActivity)getActivity()).getSupportActionBar();
+    if (actionBar != null) {
+      actionBar.setTitle(null);
+      actionBar.setDisplayUseLogoEnabled(false);
+      actionBar.setDisplayShowHomeEnabled(false);
+      actionBar.setDisplayShowTitleEnabled(false);
+      actionBar.setDisplayShowCustomEnabled(true);
+      actionBar.setBackgroundDrawable(MainActivity.getActionBarDrawable(getContext()));
+    }
+
+    TextView packageActivityTitle = (TextView) v.findViewById(R.id.bar_title);
+    packageActivityTitle.setWidth((int) getResources().getDimension(R.dimen.package_label_width));
+
+    String titleStr = pkgTarget.equals(PackageProcessor.PP_TARGET_KEYBOARDS) ?
+      getString(R.string.install_keyboard_package) :
+      getString(R.string.install_predictive_text_package);
+    if (FileUtils.isWelcomeFile(fileName)) {
+      titleStr = String.format(getString(R.string.welcome_package), pkgName);
+    }
+    packageActivityTitle.setText(titleStr);
+
+    WebView webView = (WebView) v.findViewById(R.id.packageWebView);
+    webView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
+    webView.getSettings().setJavaScriptEnabled(true);
+    webView.getSettings().setUseWideViewPort(true);
+    webView.getSettings().setLoadWithOverviewMode(true);
+    webView.getSettings().setBuiltInZoomControls(true);
+    webView.getSettings().setSupportZoom(true);
+    webView.getSettings().setTextZoom(200);
```

Comment:
I set the zoom 200% because imho the generated readme.htm (when missing) was too small at
### 100%
![khmer readme 100 percent](https://user-images.githubusercontent.com/7358010/91277465-0d7ee580-e7ad-11ea-9950-d67cf3edacba.png)

### 200%
![khmer readme](https://user-images.githubusercontent.com/7358010/91277475-12dc3000-e7ad-11ea-8830-78c8bd0af3b9.png)

Reply:
Okay, this is a mitigation that may break well-designed HTML documents. Perhaps a better mitigation would be to inject 
[meta viewport](https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag) tag into `<head>` for the file if they don't already have one, and for the future add a recommendation that keyboard developers include the viewport meta tag in their welcome.htm and readme.htm:

```html
<meta name="viewport" content="width=device-width, initial-scale=1">
```

Reply:
Something like the following might be workable (use a better injector if you have one!):

```java
    @Override
    public void onPageFinished(WebView view, String url) {
        super.onPageFinished(view, url);
        webView.loadUrl(
            "javascript:(function() { " +
               "if(!document.querySelectorAll('meta[name=viewport]').length) {"+
                 "let meta=document.createElement('meta');"+
                 "meta.name='viewport';"+
                 "meta.content='width=device-width, initial-scale=1';"+
                 "document.head.appendChild(meta);"+
               "}"+
            "})()");
    }
```